be/tool/beck.c

```diff
@@ -280,6 +284,34 @@ struct emap_action {
 	struct m0_be_emap_rec   emap_val_data;  /**< emap val data */
 };
 
+struct worker_off_info {
+	off_t              woi_offset[AO_NR];
+	uint64_t           woi_act_added[AO_NR];
+	uint64_t           woi_act_done[AO_NR];
+};
+
+struct scanner_off_info {
+	off_t    soi_offset;
+	bool     soi_scanqempty;
+	bool     soi_bnodeqempty;
+};
+
+struct part_info {
+	uint64_t           pi_act_added[AO_NR];
+	uint64_t           pi_act_done[AO_NR];
+	off_t              pi_1st_bnode_offset[AO_NR];
+};
+
+struct offset_info {
+	struct part_info        oi_pinfo;
+	struct worker_off_info *oi_offset;
```

Comment:
Can we allocate statically like op_pinfo?

Reply:
In case of partition info 'AO_NR' constant is used as array size. There is no such constant or at least max limit defined for 'tbc_workers_nr' which can be used as array size.